node/pkg/watchers/aptos/watcher.go

```diff
@@ -140,47 +330,62 @@ func (e *Watcher) Run(ctx context.Context) error {
 			if !gjson.Valid(string(body)) {
 				logger.Error("InvalidJson: " + string(body))
 				p2p.DefaultRegistry.AddErrorCount(e.chainID, 1)
-				break
-
+				continue
 			}
 
-			outcomes := gjson.ParseBytes(body)
+			// Parse the GraphQL response
+			response := gjson.ParseBytes(body)
 
-			for _, chunk := range outcomes.Array() {
-				newSeq := chunk.Get("sequence_number")
-				if !newSeq.Exists() {
-					break
-				}
+			// Extract the msg array from data.msg
+			messages := response.Get("data.msg")
+			if !messages.Exists() || len(messages.Array()) == 0 {
```

Comment:
The bounds check is worth a loud comment here because the node will panic without it.